client/app/components/packages/pas-form/edit.hbs

```diff
@@ -666,106 +616,61 @@
 
           <label>
             <strong>Description of the proposed development being facilitated by the land use actions</strong>
-            <Textarea
-              @value={{saveable-form.saveableChanges.dcpProjectdescriptionproposeddevelopment}}
-              rows="5"
-              data-test-textarea="dcpProjectdescriptionproposeddevelopment"
-            />
-            <CharacterCounter
-              @string={{saveable-form.saveableChanges.dcpProjectdescriptionproposeddevelopment}}
-              @maxlength="3000"
-            />
-            <ValidationMessage
-              @changesetError={{saveable-form.submittableChanges.error.dcpProjectdescriptionproposeddevelopment}}
-              data-test-validation-message="dcpProjectdescriptionproposeddevelopment"
+            <TextArea
+              @model={{saveable-form.saveableChanges}}
+              @property="dcpProjectdescriptionproposeddevelopment"
```

Comment:
@allthesignals can you expand a bit on what the two different approaches are?  I'm not sure I fully understand.

Reply:
Normally, we pass a direct reference of the attribute into the component:
```
            <TextArea
              @prop={{saveable-form.saveableChanges.dcpProjectdescriptionproposeddevelopment}}
```

Instead, here, we pass two things, a model and a the key name, so that we could look it up in a getter. 

I'm arguing that we pass the prop directly.

Reply:
Ahh, thank you that makes sense.  I agree that seems more straightforward, but curious to hear @godfreyyeung your thoughts on if there are challenges to this approach here.

Reply:
Passing the property name as a string allows us to easily pass that string into the test-selector attribute, and also use it to index into BOTH the changeset and changeset.error objects. Otherwise I think we would have to pass in references to both `changeset.dcp....` and `changeset.error.dcp...`

Reply:
@godfreyyeung if we use `...attributes`, we can pass the test selector from the consuming end. I think that makes fewer assumptions about how this should work.

Reply:
> I also just found it more enjoyable having to change one attribute on each input component, as opposed to three or four, when updating/fixing property names

I'm sure it was — I just worry about maintainability and "surprises". It's _more_ surprising to run across an object and a key name in a template than to pass the actual reference itself. Throughout most of our application, we consistently pass by reference, not with key names. I'd rather these components not have any context about the model itself.

Reply:
> @godfreyyeung if we use `...attributes`, we can pass the test selector from the consuming end. I think that that makes fewer assumptions about how this should work.

Though I feel like this slightly reduces the value of components. The consuming/parent template will be bloated with more repetitions of `data-test-....`

Also, is there a way to assign splattributes to multiple elements within the components?  

For example, within my <TextArea> , both the native HTML `<Textarea>` and `<ValidationMessage>` elements require data-test selectors. `data-test-textarea={{@property}}` and  `data-test-validation-message={{@property}}`, respectively. 

Also wondering if there is a way to selectively render only certain splattributes? It won't make sense to assign all the splattributes to `<ValidationMessage>`, since it doesn't need attributes like `rows`.

I can poke around..

Reply:
@godfreyyeung Cool, I think you're right. I would say that saving keystrokes is not a good enough reason in itself, but this seems justified because it'll make testing way easier. Good points.

Reply:
> @godfreyyeung my biggest issue with this was passing property by the string name instead of by reference. I went ahead and approved but could you please address those things?

> @godfreyyeung Cool, I think you're right. I would say that saving keystrokes is not a good enough reason in itself, but this seems justified because it'll make testing way easier. Good points.

Wait, sorry I'm confused -- are you maintaining that I should or shouldn't remove the passing the property by string?